src/field_theory/separable.lean

```diff
@@ -422,6 +422,9 @@ end splits
 theorem _root_.irreducible.separable [char_zero F] {f : F[X]}
   (hf : irreducible f) : f.separable :=
 begin
+  haveI : no_zero_smul_divisors ℤ F :=
+    no_zero_smul_divisors.of_algebra_map_injective ((algebra_map ℤ F).injective_int),
+  haveI : no_zero_smul_divisors ℕ F := nat.no_zero_smul_divisors ℤ F,
```

Comment:
I have just opened #14395 which should make these lines unnecessary. I suggest we wait till that is resolved. Other than this, these changes look ready.